Ed-Fi-Alliance-OSS/Fix-It-Friday
# Add JWT validation

**To test a failure:**
Click Send. It should report a HTTP 401: Invalid token. You are not authenticated

**To test a successful request**
- Copy the `sample.env` file and rename to `.env`
- Create OAUth2 Credentials at [Google Developer Console API registration](https://console.developers.google.com/).
- Replace the Client ID and Client Secret in your .env file
### For Insomnia, Switch Auth tab to OAuth2, and use the following settings:
```bash
GRANT TYPE: Implicit
AUTHORIZATION URL: https://accounts.google.com/o/oauth2/v2/auth
CLIENT ID: Use the value from your client ID
REDIRECT URL: http://localhost:3000 (this will also need to be registered with your Google app)

Under Advanced Options:
RESPONSE TYPE: ID Token
SCOPE: profile
```

- Next, click Refresh Token. You should be able to authenticate using Google and populate the ACCESS TOKEN in Insomnia.
Click Send button and it should provide a HTTP 200 result.

### For GraphQL playground, do the following:
- Generate a Google id_token. (I just used Insomnia to get the token).
- On HTTP Headers add the token as follows:
```json
{
    "Authorization" : "Bearer _eyXXXXXXXX...._"
}
```
- For a query, you can use the staff students...
```json
query Student ($key: ID!){
  studentsbystaff (staffkey: $key){
    studentkey
    sex
    schoolkey
    studentfirstname
    studentlastname
    studentmiddlename
    pictureurl
  }
}
```
And the Query Variables...
```json
{
  "key": "166930"
}
```
